docs/fireflyOverView.md

```diff
@@ -0,0 +1,219 @@
+# Firefly
```

Comment:
This file is very similar to firefly-api-overview.md.  We can rearrange the content so that it can be used by both, instead of keeping 2 separate copies.